panel/models/icon.ts

```diff
@@ -134,11 +186,76 @@ export class ToggleIcon extends Control {
   static {
     this.prototype.default_view = ToggleIconView;
 
-    this.define<ToggleIcon.Props>(({ Boolean, Nullable, String }) => ({
-      active_icon: [String, ""],
-      icon: [String, "heart"],
-      size: [Nullable(String), null],
+    this.define<ToggleIcon.Props>(({ Boolean }) => ({
       value: [Boolean, false],
     }));
   }
 }
+
+// ButtonIcon
+
+export class ButtonIconView extends ClickableIconView {
+  model: ButtonIcon;
+  value: boolean;
+
+  public *controls() { }
+
+  toggle(): void {
+    if (this.model.disabled) {
+      return;
+    }
+    const updateState = (value: boolean, disabled: boolean) => {
+      this.value = value;
+      this.model.disabled = disabled;
+      this.update_icon();
+    };
+    updateState(true, true);
+    this.model.clicks += 1;
```

Comment:
Would prefer to use a Bokeh event for this, not a property. Property changes can be silently dropped if there is a more recent property change but events will not.

Reply:
I found something similar in FileDownload, but not 100% sure if this is what you mean.

```typescript
  render(): void {
    super.render();
    this._click_listener = this.increment_clicks.bind(this)
    this.icon_view.el.addEventListener("click", this._click_listener)
  }

  increment_clicks(): void {
    if (!this.model.disabled) {
      this.model.clicks = this.model.clicks + 1
    }
  }
```

Reply:
Okay I looked at button.ts in bokeh, and I think that does what you're suggesting.